app/liquid_tags/youtube_tag.rb

```diff
@@ -34,8 +35,21 @@ def translate_url(input)
     input.split("watch?v=")[1].split("\"")[0]
   end
 
+  def translate_start_time(id)
+    time = id.split("?t=")[-1]
+    if /(\d+)m(\d+)s/.match?(time)
+      time = time.scan(/(\d+)m(\d+)s/)[0].map(&:to_i)
+      seconds = time[1] + (time[0] * 60)
+      "#{id.split('?t=')[0]}?start=#{seconds}"
+    end
+  end
+
   def valid_id?(id)
-    id.length == 11 && id =~ /[a-zA-Z0-9_-]{11}/
+    if id.length == 11
+      id =~ /[a-zA-Z0-9_-]{11}/
+    elsif id.split("?")[0].length == 11
+      id =~ /[a-zA-Z0-9_-]{11}\?start\=\d*/
```

Comment:
You can make the `?start=\d` part optional with `?` and only need one regex:
```ruby
id =~ /[a-zA-Z0-9_-]{11}[\?start\=\d*]?/ 
```

Reply:
@AntonyGarand That's what I originally had tried to do in in 03674c479a2452e8f3111a69dd666017288dccb5, but it wasn't failing to catch the invalid youtube ID `really_long_invalid_id` and I couldn't find a resolution.  
This is the regex I was using:
```ruby
id =~ /[a-zA-Z0-9_-]{11}(?:\?start\=\d*)?/
``` 
The only difference being the `?:` to match the optional portion without making it a capturing group, but it's possible that this doesn't work as I expected in Ruby regex.

Reply:
I'm happy to go with this implementation, as the logic is pretty clear. Even if more elegant, I can't imagine the pure regex solution being _easier_ to read or meaningfully cleaner.

@maestromac wanna give this all a once-over and merge if you feel good with it.

Reply:
Hey @tterb. I took a look at it and I think our `really_long_invalid_id` spec is testing something that doesn't need to be tested. After all, if an id is invalid, youtube embed will let the user know there's a mistake. Feel free to remove that spec and apply either of those 1 line regex.

Reply:
@maestromac Okay, will do.